common/docker/client.go

```diff
@@ -72,21 +78,29 @@ func (c *Client) Info() string {
 
 // AddObserver adds an observer for docker events
 func (c *Client) AddObserver(ob ContainerObserver) error {
-	events := make(chan *docker.APIEvents)
-	if err := c.AddEventListener(events); err != nil {
-		Log.Errorf("[docker] Unable to add listener to Docker API: %s", err)
-		return err
-	}
-
 	go func() {
-		for event := range events {
-			switch event.Status {
-			case "start":
-				id := event.ID
-				ob.ContainerStarted(id)
-			case "die":
-				id := event.ID
-				ob.ContainerDied(id)
+		retryInterval := InitialInterval
```

Comment:
> wouldn't we want to reset the retry interval on successful connection?

As it stands, `AddEventListener` can only return an error if you try to subscribe the same listener twice - the actual docker socket dial and `GET /events` request happen asynchronously, and the only way we learn about either of them failing is via event channel closure.

Consequently, in this case, 'successful connection' equates to the `for ... range events` _blocking_, which does not give us a handy place to reset the interval. I did give it some thought, the best I could come up with was to record the amount of time we spend in the event loop and if it exceeds the retryInterval when the channel closes we reset. Thoughts?